packages/trading-widget/src/trading-widget/components/withdraw/init-step/input-group/withdraw-section/all-assets-composition-table/all-assets-composition-table.tsx

```diff
@@ -0,0 +1,128 @@
+import { ArrowTopRightOnSquareIcon } from '@heroicons/react/24/outline'
+import classNames from 'classnames'
+
+import {
+  FLATMONEY_COLLATERAL_SYMBOL_MAP,
+  FLAT_MONEY_UNIT_LINK,
+  SYNTHETIX_V3_VAULTS_WITHDRAW_ASSET_SYMBOL_MAP,
+} from 'core-kit/const'
+import type { PoolCompositionWithFraction } from 'core-kit/types'
+import {
+  getFlatMonetUnitSymbol,
+  isFlatMoneyLeveragedRethAsset,
+  isSynthetixV3Asset,
+} from 'core-kit/utils'
+import { Skeleton, TokenBadge } from 'trading-widget/components/common'
+
+import {
+  AssetsCompositionDisclosure,
+  AssetsCompositionTable,
+} from 'trading-widget/components/common/meta/assets-composition-disclosure/assets-composition-disclosure'
+import type { AllAssetsCompositionTableProps } from 'trading-widget/components/withdraw/init-step/input-group/withdraw-section/all-assets-composition-table/all-assets-composition-table.hooks'
+import { useAllAssetsCompositionTable } from 'trading-widget/components/withdraw/init-step/input-group/withdraw-section/all-assets-composition-table/all-assets-composition-table.hooks'
+import { WithdrawExplanationTip } from 'trading-widget/components/withdraw/init-step/input-group/withdraw-section/all-assets-composition-table/withdraw-explanation-tip'
+
+export const AllAssetsCompositionTable = ({
+  className,
+  showFraction = true,
+  iconSize,
+  showAllAsset = false,
+}: AllAssetsCompositionTableProps) => {
+  const {
+    visibleAssets,
+    hiddenAssets,
+    chainId,
+    showUnitWithdrawalTip,
+    address,
+  } = useAllAssetsCompositionTable(showAllAsset)
+
+  const renderRow = ({
```

Comment:
move out of component if possible

Reply:
wrapped in `useCallback` as it uses some props